submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt

```diff
@@ -78,35 +78,58 @@ class SubmissionRequestCleanIndexer(
         val deprecatedIdx = AtomicInteger(0)
         val deprecatedPageTabIdx = AtomicInteger(0)
 
-        fun fileUpdate(
-            idx: AtomicInteger,
-            file: ExtFile,
-            status: RequestFileStatus,
-        ) = SubRqtFile(new, idx.incrementAndGet(), file, status, true)
+        fun indexRequestFile(
+            file: PersistedExtFile,
+            isPageTab: Boolean,
+        ): SubmissionRequestFile? =
+            when (newFiles.findMatch(file, isPageTab)) {
+                MatchType.CONFLICTING -> {
+                    conflictIdx.incrementAndGet()
+                    SubmissionRequestFile(new, file, CONFLICTING, true)
+                }
+
+                MatchType.CONFLICTING_PAGE_TAB -> {
+                    conflictPageTabIdx.incrementAndGet()
+                    SubmissionRequestFile(new, file, CONFLICTING_PAGE_TAB, true)
+                }
+
+                MatchType.DEPRECATED -> {
+                    deprecatedIdx.incrementAndGet()
+                    SubmissionRequestFile(new, file, DEPRECATED, true)
+                }
+
+                MatchType.DEPRECATED_PAGE_TAB -> {
+                    deprecatedPageTabIdx.incrementAndGet()
+                    SubmissionRequestFile(new, file, DEPRECATED_PAGE_TAB, true)
+                }
+
+                MatchType.REUSED -> {
+                    when {
+                        current.released && new.released -> SubmissionRequestFile(new, file, RELEASED, false)
+                        current.released.not() && new.released.not() -> SubmissionRequestFile(new, file, COPIED, false)
```

Comment:
Couldn't these two be simplified with `current.released.not() -> SubmissionRequestFile(new, file, COPIED, false)` ?